external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -516,9 +506,110 @@ pub fn build_member_map(
         }
         assert!(all_members.insert(mident, members).is_none());
     }
+
+    if let Some(pre_compiled_lib) = pre_compiled_lib {
+        all_members.extend(pre_compiled_lib.iter().filter_map(|(mident, minfo)| {
+            if !prog.modules.contains_key(mident) {
+                Some((*mident, pre_compiled_resolved_members(*mident, &minfo.info)))
+            } else {
+                None
+            }
+        }));
+    }
+
     all_members
 }
 
+/// Convert a pre-compiled module info to a map of resolved members.
+fn pre_compiled_resolved_members(
```

Comment:
Is it possible to avoid the near-direct duplication of this code from above? Could we write a trait for info that hands back the appropriate definitions and write this as
```
fn compute_named_module_members<M: ResolvableModule>(
    mident: ModuleIdent,
    pre_compiled_module_info: &M,
) -> BTreeMap<Symbol, ResolvedModuleMember> {
```